libbeat/docs/outputconfig.asciidoc

```diff
@@ -215,53 +213,112 @@ for more information about the environment variables.
 
 The index name to write events to. The default is
 +"{beatname_lc}-%\{[beat.version]\}-%\{+yyyy.MM.dd\}"+ (for example,
-+"{beatname_lc}-{version}-2017.04.26"+). If you change this setting, you also
++"{beatname_lc}-{version}-{localdate}"+). If you change this setting, you also
 need to configure the `setup.template.name` and `setup.template.pattern` options
-(see <<configuration-template>>).
+(see <<configuration-template>>). If you are using the pre-built Kibana
+dashboards, you also need to set the `setup.dashboards.index` option (see
+<<configuration-dashboards>>).
+
+You can set the index dynamically by using a format string to access any event
+field. For example, this configuration uses a custom field, `fields.log_type`,
+to set the index: 
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["http://localhost:9200"]
+  index: "%\{[fields.log_type]\}-%\{[beat.version]\}-%\{+yyyy.MM.dd}\" <1>
+------------------------------------------------------------------------------
+
+<1> We recommend including `beat.version` in the name to avoid mapping issues
+when you upgrade.
+
+With this configuration, all events with `log_type: normal` are sent to an 
+index named +normal-{version}-{localdate}+, and all events with
+`log_type: critical` are sent to an index named
++critical-{version}-{localdate}+.
+
+TIP: To learn how to add custom fields to events, see the
+<<libbeat-configuration-fields,`fields`>> option.  
+
+See the <<indices-option-es,`indices`>> setting for other ways to set the index
+dynamically.
 
-If you are using the pre-built Kibana dashboards,
-you also need to set the `setup.dashboards.index` option (see <<configuration-dashboards>>).
 
 ifdef::deprecate_dashboard_loading[]
 
 deprecated[{deprecate_dashboard_loading}]
 
 endif::[]
 
+[[indices-option-es]]
 ===== `indices`
 
-Array of index selector rules supporting conditionals, format string
-based field access and name mappings. The first rule matching will be used to
-set the `index` for the event to be published. If `indices` is missing or no
-rule matches, the `index` field will be used.
+An array of index selector rules. Each rule specifies the index to use for
+events that match the rule. During publishing, {beatname_uc} uses the first
+matching rule in the array. Rules can contain conditionals, format string-based
+fields, and name mappings. If the `indices` setting is missing or no rule
+matches, the <<index-option-es,`index`>> setting is used.
 
 Rule settings:
 
-*`index`*: The index format string to use. If the fields used are missing, the rule fails.
+*`index`*:: The index format string to use. If this string contains field
+references, such as `%{[fields.name]}`, the fields must exist, or the rule fails. 
 
-*`mapping`*: Dictionary mapping index names to new names
+*`mapping`*:: A dictionary that takes the value returned by `index` and maps it
```

Comment:
s/mapping/mappings/

Reply:
Good catch! I fixed it in the code, but not the descriptions.